apollo-mutiny-support/src/main/java/com/apollographql/apollo/mutiny/MutinyApollo.java

```diff
@@ -0,0 +1,181 @@
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.cache.normalized.ApolloStoreOperation;
+import com.apollographql.apollo.exception.ApolloException;
+import com.apollographql.apollo.internal.subscription.ApolloSubscriptionTerminatedException;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.subscription.BackPressureStrategy;
+import org.jetbrains.annotations.NotNull;
+
+import static com.apollographql.apollo.api.internal.Utils.checkNotNull;
+
+/**
+ * The MutinyApollo class provides methods for converting ApolloCall, ApolloPrefetch and ApolloWatcher types to Mutiny sources.
+ */
+public class MutinyApollo {
+
+  private MutinyApollo() {
+    throw new AssertionError("This class cannot be instantiated");
+  }
+
+  /**
+   * Converts an {@link ApolloQueryWatcher} to an asynchronous Uni.
+   *
+   * @param watcher the ApolloQueryWatcher to convert.
+   * @param <T> the value type
+   * @return the converted Uni
+   * @throws NullPointerException if watcher == null
+   */
+  @NotNull
+  public static <T> Uni<Response<T>> from(@NotNull final ApolloQueryWatcher<T> watcher) {
+    checkNotNull(watcher, "watcher == null");
+    return Uni.createFrom().emitter(emitter -> {
+      ApolloQueryWatcher<T> clone = watcher.clone();
+      emitter.onTermination(clone::cancel);
+      clone.enqueueAndWatch(new ApolloCall.Callback<T>() {
+        @Override public void onResponse(@NotNull Response<T> response) {
+          emitter.complete(response);
+        }
+
+        @Override public void onFailure(@NotNull ApolloException e) {
+          emitter.fail(e);
+        }
+      });
+    });
+  }
+
+  /**
+   * Converts an {@link ApolloCall} to an {@link Uni}. The number of emissions this Uni will have is based on the {@link
+   * com.apollographql.apollo.fetcher.ResponseFetcher} used with the call.
+   *
+   * @param call the ApolloCall to convert
+   * @param <T> the value type.
+   * @return the converted Uni
+   * @throws NullPointerException if originalCall == null
+   */
+  @NotNull
+  public static <T> Uni<Response<T>> from(@NotNull final ApolloCall<T> call) {
+    checkNotNull(call, "call == null");
+    return Uni.createFrom().emitter(emitter -> {
+          ApolloCall<T> clone = call.toBuilder().build();
+          emitter.onTermination(clone::cancel);
+          clone.enqueue(new ApolloCall.Callback<T>() {
+                          @Override public void onResponse(@NotNull Response<T> response) {
+                            emitter.complete(response);
+                          }
+
+                          @Override public void onFailure(@NotNull ApolloException e) {
+                            emitter.fail(e);
+                          }
+
+                          @Override public void onStatusEvent(@NotNull ApolloCall.StatusEvent event) {
+                            if (event == ApolloCall.StatusEvent.COMPLETED) {
+                              emitter.complete(null);
```

Comment:
If the subscriber already got the response, this call will be dropped. I believe it's ok but better double-check.

Reply:
In Rxjava we have: 

```java
          if (event == ApolloCall.StatusEvent.COMPLETED && !emitter.isDisposed()) {
            emitter.onComplete();
          }
```
I don't know how to achieve equivalent check in Mutiny.

Reply:
You may register a callback for _terminatIon_: https://javadoc.io/static/io.smallrye.reactive/mutiny/0.18.1/io/smallrye/mutiny/subscription/UniEmitter.html

Reply:
Other than that agreed with @cescoffier